x-pack/plugins/observability_solution/infra/server/lib/helpers/get_infra_metrics_client.ts

```diff
@@ -42,6 +48,13 @@ export async function getInfraMetricsClient({
           ...searchParams,
           ignore_unavailable: true,
           index: metricsIndices,
+          body: {
+            ...searchParams.body,
+            query: getDataTierFilterCombined({
```

Comment:
I decided to forego it for now, because I think the object it is returning might not be entirely correct. Or I might have been using it wrong, but either way. Nested object merging is painful 😅